awx/conf/settings.py

```diff
@@ -352,7 +352,6 @@ def _preload_cache(self):
         self.cache.set_many(settings_to_cache, timeout=SETTING_CACHE_TIMEOUT)
 
     def _get_local(self, name, validate=True):
-        self.__clean_on_fork__()
```

Comment:
This gets called **a lot**, and I think the frequency of this call explains the racy nature of this bug.